### PR TITLE
Rename effectiveness

### PR DESCRIPTION
**Pull request by davidsantiagoquevedo**
This PR responds to #83.

### Main Changes

- The function `effectiveness` was renamed to `estimate_vaccineff`.
- The S3 class object `effectiveness` was renamed to `vaccineff`.
- All tests and examples were updated to reflect these changes.
- In examples and tests, the default assignment for creating `vaccineff` objects was set to `ve <- estimate_vaccineff()` to avoid confusion with the package name.

PR #95 was merged to avoid conflicts and will be reviewed separately. The changes specific to this PR are in commit 5d1620fc8acfc06842158554a66f78be979ab851.

### CODE REVIEW

**Review by chartgerink — CHANGES_REQUESTED** [review]
Thanks @davidsantiagoquevedo 🙌  The rename seems straightforward and not problematic. Let me explain why I am requesting changes nonetheless.

From my point of view, you now effectively blocked yourself from merging this until #95 is merged. The merge you did now creates a convoluted history. I cannot approve just https://github.com/epiverse-trace/vaccineff/commit/5d1620fc8acfc06842158554a66f78be979ab851 without effectively approving the changes irrelevant to the "Rename effectiveness".  As a result, I do not approve the PR in this form. 

If you rebase the PR to merge into `77-estimate-effectiveness-at` I can approve, and it'll be included when #95 is merged. If you are worried about conflicts with another branch in the future, I would recommend to create the PR on that branch instead of main, to prevent this convolution from happening. 

It is a difficult issue, so if you want me to expand some more to further illustrate my thinking, let me know and I'll try to explain another way.

**Comment by davidsantiagoquevedo** [comment]
Thanks for your review @chartgerink. I had to do it this way because I made major changes in the structure of  `estimate_vaccineff` (before `effectiveness`) and other important functions in #95, and I needed them to make sense of this new feature. I will wait until I merge #95 to proceed with merging this.

File: R/estimate_vaccineff.R
Comment (chartgerink) [inline R/estimate_vaccineff.R:45]:
```suggestion
#' ve <- estimate_vaccineff(vaccineff_data, 90)
```

Mind the style 😊

File: R/estimate_vaccineff.R
Comment (chartgerink) [inline R/estimate_vaccineff.R:125]: Could you confirm this is no longer relevant? Goes beyond renaming so double-check 😊

**Review by chartgerink — APPROVED** [review]
Thanks for updating 😊 Left two super minor comments, for your consideration, but not blocking.

File: R/estimate_vaccineff.R
Comment (davidsantiagoquevedo) [reply R/estimate_vaccineff.R:125]: Yes, It's not necessary anymore, since `at` must be provided to make the estimation fully meaningful. But it's a good sanity check, we also hesitated a bit about this within the team :)

**Pull request merged**
